test/Unit/App/Attribute/ResourceDeprecationTest.php

```diff
@@ -46,4 +47,40 @@ public function testValidDatePassesValidation(): void
         $this->assertSame('test-link', $attribute->link);
         $this->assertSame('test-deprecation-reason', $attribute->deprecationReason);
     }
+
+    public function testToArray(): void
+    {
+        $class = new #[ResourceDeprecation(
+            sunset: '2038-01-01',
+            link: 'test-link',
+            deprecationReason: 'test-deprecation-reason',
+            rel: 'test-rel',
+            type: 'test-type',
+        )] class {
+        };
+
+        $reflectionClass = new ReflectionClass($class);
+        $attributes      = $reflectionClass->getAttributes(ResourceDeprecation::class);
+
+        $this->assertIsArray($attributes);
+        $attribute = $attributes[0]->newInstance();
+
+        $array = $attribute->toArray();
+
+        $this->assertIsArray($array);
```

Comment:
```suggestion
```
psalm: RedundantCondition: Type array<array-key, mixed> for $array is always array<array-key, mixed>